### ISSUE TITLE
False negative - tuple unpacking & empty list

### ISSUE THREAD

**Issue by mbyrnepr2** [PyCQA/pylint#5707]
### Bug description

As mentioned in https://github.com/PyCQA/pylint/issues/5671, the Pylint 2.13 logic introduced the false negative with tuple unpacking.

The following example would have emitted an `unbalanced-tuple-unpacking` warning before the change of https://github.com/PyCQA/pylint/pull/5346:

mytest.py:
```
def f(x):
    l = []
    for i in x:
        l.append(i)
    return l

a, b, c, d, e, f, g = f('12')
```

In 2.13 it does not emit any warning. Would be nice to fix the unintended change to the behaviour introduced in https://github.com/PyCQA/pylint/pull/5346.

### Command used

```shell
pylint mytest.py
```

### Pylint output

```shell
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Expected behavior

mytest.py:7:0: W0632: Possible unbalanced tuple unpacking with sequence defined at line 2: left side has 7 label(s), right side has 0 value(s) (unbalanced-tuple-unpacking)

### Pylint version

```shell
pylint 2.13.0-dev0
astroid 2.9.3
Python 3.10.0b2 (v3.10.0b2:317314165a, May 31 2021, 10:02:22) [Clang 12.0.5 (clang-1205.0.22.9)]
```
